components/core/src/clp/streaming_archive/writer/Archive.cpp

```diff
@@ -242,6 +246,10 @@ void Archive::close() {
 
     m_metadata_db.close();
 
+    if (m_use_single_file_archive) {
+        create_single_file_archive();
```

Comment:
One concern:

Now we are generating a normal archive, writing its metadata to the global metadata database and then create a single file archive by combining the files.

We haven't decided the metadata to be written to global metadataDB for sfa yet, but if it would be different from normal archive, would this be an issue?

Reply:
Sure.